### PR TITLE
feat(skillopt): wire pi + generalize skill detection + CodeQL fix

### PR DESCRIPTION
**Pull request by kaghni**
Combined follow-up to the SkillOpt release (3 items in one PR per request).

## 1. Other agents — generalize detection + wire pi
- `armSkillOptOnSkillUse` now arms on EITHER a `Skill` tool (claude) OR a `read` of `…/skills/<name--author>/SKILL.md` (agents with no first-class Skill tool). markSkillPending still gates org-shape + pull manifest.
- **pi wired**: the extension arms on a `tool_result` SKILL.md read and, on the next user prompt (the reaction), spawns the bundled `skillopt-worker` (`agent=pi`). Logic inlined into the raw-.ts extension (zero-dep); `HIVEMIND_SKILLOPT_*` env names are the cross-process contract. esbuild builds + install-pi installs the worker for pi.
- **codex still blocked**: its `exec_command` read of SKILL.md isn't captured (0.136.0 PostToolUse gap) — nothing to detect yet. Filed as a prerequisite.

## 2. CodeQL — command-injection fix
`resolveAgentBin` did `execSync(\`command -v ${AGENT_CMD[agent] ?? agent}\`)`; `agent` traces to `HIVEMIND_SKILLOPT_AGENT` and the `?? agent` fallback let an arbitrary value reach a shell. Now resolves KNOWN agents only by walking $PATH in Node (no shell/subprocess). Verified: known agent → bin path, injection payload → undefined.

## 3. Coverage
- skillopt-hook: SKILL.md-read arming + ref extraction (cherry-picked with the generalization).
- pi-extension-source: 5 new source-level guards for the pi wiring (runtime-loaded → source is the bundle guard).

tsc clean (changed files); 69 tests green.

## ⚠️ Before merge
**pi wiring needs a live pi-session E2E** — the extension is loaded + compiled by pi's runtime, so it's only source-tested here. Same pattern we used for claude (which is now live-verified). Don't merge until a real pi session has been driven through arm→react→publish.

## Summary by CodeRabbit

* **New Features**
  * SkillOpt: session-aware detection of org-skill usage that can create per-session judgment windows and spawn background judgment workers triggered by subsequent user reactions.

* **Improvements**
  * SkillOpt worker bundles are built and installed for Codex, Hermes, and Pi so judgments run where applicable.
  * More robust agent-binary resolution avoiding shell subprocesses.
  * SkillOpt respects a runtime kill-switch and recursion guard.

* **Tests**
  * Expanded test coverage for wiring, detection, env-contracts, arming/reacting flows, and edge cases.

### CODE REVIEW

**Comment by kaghni** [comment]
@coderabbitai review

File: pi/extension-source/hivemind.ts [hunk 1]
```diff
@@ -466,6 +466,85 @@ function runAutopullWorker(): void {
...
+// .ts with zero non-builtin deps (it can't import the skillify trigger). pi has no first-class
+// `Skill` tool — it USES a skill by READING its SKILL.md — so we arm on a tool_result whose
+// path is .../skills/<name--author>/SKILL.md, then on the next user prompt (the reaction) spawn
+// the bundled skillopt-worker to judge + improve. Env-var names are the cross-process contract
+// with the worker (SKILLOPT_ENV in src/skillify/skillopt-env.ts) — kept as literals here since
+// the extension can't import. Fully swallowed; never blocks pi. Both call sites sit AFTER the
+// handler's captureEnabled check, so the worker's own pi-judge subprocess (HIVEMIND_CAPTURE=false)
+// can't re-arm/re-react — that's the recursion guard.
+const PI_SKILLOPT_WORKER_PATH = join(homedir(), ".pi", "agent", "hivemind", "skillopt-worker.js");
+const SKILLOPT_PENDING_DIR = join(homedir(), ".deeplake", "state", "skillify", "skillopt", "pending");
```
[the remark this answers is not in the document]
Comment (kaghni) [reply pi/extension-source/hivemind.ts]: Fixed in a6490225 — pi's pending dir now mirrors getStateDir()'s contract: a non-empty (trimmed) HIVEMIND_STATE_DIR overrides the default ~/.deeplake/state/skillify root, so pi state co-locates with the rest of Skillify.

File: pi/extension-source/hivemind.ts [hunk 2]
```diff
@@ -466,6 +466,85 @@ function runAutopullWorker(): void {
...
+  const safe = sessionId.replace(/[^A-Za-z0-9_-]/g, "_").slice(0, 200);
+  return join(SKILLOPT_PENDING_DIR, `${safe}.json`);
+}
+
+/** tool_result: pi read an org skill's SKILL.md → open a K-message judgment window. */
+function skilloptArm(sessionId: string, toolInput: any, toolCallId: unknown): void {
+  try {
+    if (process.env.HIVEMIND_SKILLOPT_DISABLED === "1") return;
+    const ref = skilloptRefFromPath(toolInput?.path ?? toolInput?.file ?? toolInput?.filePath);
+    if (!ref) return;
```
[the remark this answers is not in the document]
Comment (kaghni) [reply pi/extension-source/hivemind.ts]: Fixed in a6490225 — skilloptArm now takes the toolName and arms ONLY on read tools (`/^read/i`), so an edit/write of a SKILL.md can't open a judgment window. Covered by the pi-extension-source 'read tools only' guard.

File: src/hooks/codex/pre-tool-use.ts [hunk 3]
```diff
@@ -368,6 +369,9 @@ export async function processCodexPreToolUse(
 /* c8 ignore start */
 async function main(): Promise<void> {
   const input = await readStdin<CodexPreToolUseInput>();
+  // SkillOpt: codex USES an org skill by shelling a read of its SKILL.md — arm the judgment
+  // window on that command. Fully swallowed; never affects the tool decision below.
+  armSkillOptOnSkillUse(input.session_id, input.tool_name, input.tool_input, input.tool_use_id);
   const decision = await processCodexPreToolUse(input);
```
[the remark this answers is not in the document]
Comment (kaghni) [reply src/hooks/codex/pre-tool-use.ts:378]: Fixed in a4b91381 — the armSkillOptOnSkillUse call in codex main() is now wrapped in try/catch so a throw can't short-circuit processCodexPreToolUse (whose top-level catch exits 0). Fail-closed for the SkillOpt side-effect.

File: src/skillify/skill-invocations.ts [hunk 4]
```diff
@@ -44,13 +44,36 @@ export function parseMessage(m: unknown): ParsedMsg | null {
...
-  const skill = (input as { skill?: unknown })?.skill;
-  return typeof skill === "string" && skill.length > 0 ? skill : null;
+  if (typeof input === "string") { try { input = JSON.parse(input); } catch { input = msg.tool_input; } }
+  if (msg.tool_name === "Skill") {
+    const skill = (input as { skill?: unknown })?.skill;
+    return typeof skill === "string" && skill.length > 0 ? skill : null;
+  }
+  const io = input as { path?: unknown; command?: unknown } | undefined;
+  return pathToSkillRef(io?.path) ?? pathToSkillRef(io?.command);
 }
```
[the remark this answers is not in the document]
Comment (kaghni) [reply src/skillify/skill-invocations.ts:77]: Fixed in 0bde4300 — the listSkillInvocations prefilter now matches a `"Skill"` tool_call OR a `/SKILL.md` path, so read/shell SKILL.md invocations aren't dropped before invokedSkillRef. Test asserts both branches.

**Comment by kaghni** [comment]
@coderabbitai review

**Review by efenocchi — APPROVED** [review]
